src/content/1.7/modules/creation/module-translation/new-system.md

```diff
@@ -0,0 +1,302 @@
+  - Symbols like dots, dashes and underscores are allowed (but discouraged).
+  - If you want to use previously-generated classic translation dictionary files, you need to follow a [specific naming convention for backwards compatibility](#backwards-compatibility).  
+     Otherwise, consider using the [Native module conventions][native-module-conventions] (_"Admin"_ or _"Shop"_).
+
+#### Backwards compatibility
+
+If you want your module to be compatible with previously-generated [classic translation dictionary files]({{< ref "classic-system.md" >}}), then the third component of the Translation domain must be set to the name of the file where the wording is used, respecting the following rules:
+  
+  - The first letter must be in upper case, and the rest in lower case
+  - If the file extension is `.tpl`, the extension must be removed
+  - If the file name ends in "controller", that part must be removed as well.       
+
+##### Examples
+ 
+Assuming your module is called "my_module":
+
+File where the wording is used | Expected translation domain
+--- | ---
+my_module.php | Modules.Mymodule.My_module.php
+SomeFile.php | Modules.Mymodule.Somefile.php
+a_certain_template.tpl | Modules.Mymodule.A_certain_template
+ps_somefile.tpl | Modules.Mymodule.Ps_somefile
+another-template.html.twig | Modules.Mymodule.Another-template.html.twig
+
+You can find more examples in the [test fixtures for DomainHelper][domainhelper].
+
+### PHP files
+
+In PHP files, translation is performed using the module's `trans()` method.
+
+This method takes four parameters:
+     
+1. `$id` – The wording you want to translate.
+2. `$parameters` – An array of replacements, if any. ([Learn more about translation placeholders](https://symfony.com/doc/3.4/components/translation/usage.html#component-translation-placeholders)).
+3. `$domain` – The translation domain for that wording, as explained above.
+4. `$locale` – (optional) The locale identifier (eg. "en-US") if you want to translate in a different language than the current one.
+
+Now let's see some examples on how to use it.
+
+#### Module's main class
+
+When translating wordings in the module's main class, since it extends the `Module` class, you can simply call `$this->trans()`.
+
+```php
+// file: mymodule.php
+
+class mymodule extends Module
+{
+    public function __construct()
+    {
+        $this->displayName = $this->trans('My module', [], 'Modules.Mymodule.Mymodule');
+        $this->description = $this->trans('Description of my module.', [], 'Modules.Mymodule.Mymodule');
+    }
+}
+```
+
+Since the module is called MyModule, the translation domain should be `Modules.Mymodule.Mymodule`. The third part matches the file name, which is also "mymodule".
+
+#### Module controllers
+
+`ModuleAdminController` and `ModuleFrontController` can access the module instance via the `$this->module` property.
+
+```php
+// file: controllers/front/something.php
```

Comment:
question: what will happens if I have another file named something.php in my module?

Reply:
AFAIK they will both treated as if they were the same file